### PR TITLE
feat: Anchor onClick

### PR DESCRIPTION
**Pull request by mushan0x0**
* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [x] Add unit tests for the feature.

### CODE REVIEW

**Comment by afc163** [comment]
ci broken

File: components/anchor/Anchor.tsx [hunk 1]
```diff
@@ -101,7 +105,15 @@ export interface AnchorDefaultProps extends AnchorProps {
   getContainer: () => AnchorContainer;
 }
 
-export default class Anchor extends React.Component<AnchorProps, any> {
+export interface AntAnchor {
+  registerLink: (link: string) => void;
+  unregisterLink: (link: string) => void;
+  activeLink: string | null;
+  scrollTo: (link: string) => void;
+  propClick?: (e: React.MouseEvent<HTMLElement>, link: { title: React.ReactNode, href: string }) => void;
```
Comment (afc163) [inline components/anchor/Anchor.tsx]: How about `onClick`?

Comment (mushan0x0) [reply components/anchor/Anchor.tsx]: 我觉得可以

**Comment by mushan0x0** [comment]
ci又失败了，我觉得不是代码的问题

**Pull request merged**